### PR TITLE
VM Tidy Up

### PR DESCRIPTION
**Pull request by jasonwilliams**
- comment out `pushUndefined` and `Return` https://github.com/boa-dev/boa/discussions/1607#discussioncomment-1416211
- Move call frame to its own file (we may end up adding more code here later)
- heading for code blocks needed to be on its own line
- Show the difference between the VM starting up and a new Call frame being ran
- add trace to VSCode launch

### CODE REVIEW

File: boa/src/bytecompiler.rs [hunk 1]
```diff
@@ -868,8 +868,8 @@ impl ByteCompiler {
 
         compiler.code_block.params = paramaters.to_owned().into_boxed_slice();
 
-        compiler.emit(Opcode::PushUndefined, &[]);
-        compiler.emit(Opcode::Return, &[]);
+        // compiler.emit(Opcode::PushUndefined, &[]);
+        // compiler.emit(Opcode::Return, &[]);
```
Comment (HalidOdat) [inline boa/src/bytecompiler.rs:872]: we can't do this without knowing if the last node is a return statement. otherwise the function will never return.

Comment (jasonwilliams) [reply boa/src/bytecompiler.rs]: I've added it back

**Review by raskad — APPROVED** [review]
Looks good to me

File: boa/src/vm/mod.rs [hunk 2]
```diff
@@ -521,11 +509,17 @@ impl Context {
         const OPERAND_COLUMN_WIDTH: usize = COLUMN_WIDTH;
         const NUMBER_OF_COLUMNS: usize = 4;
 
+        let msg = if !self.vm.frame().exit_on_return {
+            " Call Frame "
+        } else {
+            " VM Start"
+        };
+
```
Comment (RageKnify) [inline boa/src/vm/mod.rs:517]: Can you swap the order on this? I understand `Call Frame` is the more common one but I think having simpler logic trumps that.
```suggestion
        let msg = if self.vm.frame().exit_on_return {
            " VM Start"
        } else {
            " Call Frame "
        };

```

**Review by RageKnify — CHANGES_REQUESTED** [review]

**Review by RageKnify — APPROVED** [review]

**Pull request merged**